### PR TITLE
Fixed rack form silently dropping custom field values

### PR DESCRIPTION
**Pull request by gsnider2195**
# Closes: #4396
# What's Changed

Small bug fix in the form clean method. This should probably be back ported to 1.6 

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/dcim/forms.py [hunk 1]
```diff
@@ -459,7 +459,7 @@ def clean(self):
                         "would conflict with same-named devices in this rack."
                     }
                 )
-        return cleaned_data
+        return super().clean()
```
Comment (glennmatthews) [inline nautobot/dcim/forms.py:462]: In a lot of cases `form.clean()` doesn't actually return anything, as the "outcome" of the function is just to populate `self.cleaned_data`. What happens if we just remove this line?

Comment (glennmatthews) [reply nautobot/dcim/forms.py:462]: I think the actual issue is that we were never calling `super.clean()` in this function.

**Review by glennmatthews — APPROVED** [review]

File: nautobot/dcim/forms.py (hunk 1 above)
Comment (gsnider2195) [reply nautobot/dcim/forms.py:462]: Yep. The `CustomFieldModelFormMixin` clean method is what is actually taking the form custom field data and adding it to the instance. The `RackForm` clean method didn't exist. When it was added without `super().clean()` it introduced this bug.

**Comment by glennmatthews** [comment]
also :+1: on backporting this fix to 1.6.

**Pull request merged**